src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs

```diff
@@ -150,16 +150,16 @@ public static bool TryFind(ReadOnlySpan<char> pemData, out PemFields fields)
                 return true;
 
                 NextAfterLabel:
-                if (preebEndIndex <= 0)
+                if (preebEndIndex <= areaOffset)
                 {
                     // We somehow ended up in a situation where we will advance
-                    // 0 or -1 characters, which means we'll probably end up here again,
-                    // advancing 0 or -1 characters, in a loop. To avoid getting stuck,
+                    // backward or not at all, which means we'll probably end up here again,
+                    // advancing backward, in a loop. To avoid getting stuck,
                     // detect this situation and return.
                     fields = default;
                     return false;
                 }
-                areaOffset += preebEndIndex;
+                areaOffset = preebEndIndex;
```

Comment:
The `+=` used to be right back when an earlier implementation maintained two spans, one for  the "real" user input and another for where we are currently looking. However that was refactored away and all of the offsets used are now absolute values in to the original PEM string. So we shouldn't accumulate the offset anymore.